src/utp_socket_manager.cpp

```diff
@@ -255,37 +255,48 @@ namespace libtorrent
 
 	void utp_socket_manager::writable()
 	{
-		std::vector<utp_socket_impl*> stalled_sockets;
-		m_stalled_sockets.swap(stalled_sockets);
-		for (std::vector<utp_socket_impl*>::iterator i = stalled_sockets.begin()
-			, end(stalled_sockets.end()); i != end; ++i)
+		if (m_in_writable) return;
```

Comment:
I don't think this is necessary anymore. It's also not obvious that this is correct (on the off-chance that this would actually get called again during handling). I would prefer to have as simple code as possible, and just not expect re-entrancy